### PR TITLE
Feature/state based authorization

### PR DESCRIPTION
**Pull request by bramirez96**
# Description

This branch fixes a potential security vulnerability/unintended interaction where users did not have to select a profile to enter our site after login. A wrapper component has been added to the SecureRoute component from Okta that consumes state and checks to make sure a user is both logged in AND has selected a profile before they are able to access any secure pages on our site, as was originally intended.

This PR also includes a new action that will clear the state of both parent and child dashboards so that we can add a "change user" functionality to the menus on both dashboards, as well as a small fix that autofocuses the PIN input after clicking on a profile.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [ ] Yes
- [ ] No
- [x] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by JLC6290 — APPROVED** [review]

**Review by Nick-Ohman — APPROVED** [review]
This looks great! Nice job with store and the tests, thats noted.

File: src/__tests__/LandingPage.test.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
 import { LandingPage } from '../components/pages/LandingPage';
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+
+import configureStore from 'redux-mock-store';
+const mockStore = configureStore([]);
+const store = mockStore();
+
+const Component = () => {
```
Comment (Schrese) [inline src/__tests__/LandingPage.test.js:10]: I really like this pattern!

File: src/components/common/Header.js [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
-import { Link } from 'react-router-dom';
+import { Link, useHistory } from 'react-router-dom';
 import { Button, Dropdown, Menu } from 'antd';
 import { MenuOutlined } from '@ant-design/icons';
+import { connect } from 'react-redux';
+import { global } from '../../state/actions';
+import { useOktaAuth } from '@okta/okta-react/dist/OktaContext';
 
-const ChildMenu = ({ authService, ...props }) => {
+const ChildMenu = props => {
```
Comment (Schrese) [inline src/components/common/Header.js:10]: Would we want to break these out into separate components at some point?

File: src/__tests__/LandingPage.test.js (hunk 1 above)
Comment (bramirez96) [reply src/__tests__/LandingPage.test.js:10]: 😄

**Review by Schrese — APPROVED** [review]
Looking great! Thanks for doing this!

File: src/components/common/Header.js (hunk 2 above)
Comment (Schrese) [reply src/components/common/Header.js:10]: good point. thanks

**Pull request merged**